okhttp/src/main/java/okhttp3/EventListener.java

```diff
@@ -269,11 +270,10 @@ public void responseBodyStart(Call call) {
    *
    * <p>This method is always invoked after {@link #requestBodyStart(Call)}.
    *
-   * <p>{@code throwable} will be null in the case of a successful attempt to send the body.
-   *
-   * <p>{@code throwable} will be non-null in the case of a failed attempt to send the body.
+   * @param bytesRead the length in bytes of the body read, including partial success.
+   * @param throwable null if response body was successfully received, non-null otherwise.
```

Comment:
With `Callback` we report only `IOException` failures; everything else is a crash. Should we do likewise for event listeners? I suspect it might simplify our code to not attempt to send say, a `NullPointerException` caused by OkHttp crashing to the event listener.

Or in other words, if there’s an exception that isn’t an IOException our invariants are broken anyway, and we shouldn’t attempt to provide nice errors when we know we’re broken.

(This isn’t for this PR, but something to contemplate for a follow-up).

Reply:
I like that suggestion a lot.  Throwable is awkward and we can differentiate expected exceptions from things are fundamentally broken.  It also matches the behaviour for call execute/enqueue.

https://square.github.io/okhttp/3.x/okhttp/okhttp3/Callback.html#onFailure-okhttp3.Call-java.io.IOException-